### DISCUSSION TITLE
Hilfe bei shelly2 pro PM

### DISCUSSION THREAD

**Discussion by msvintaudio** [camueller/SmartApplianceEnabler#488] [Installation und Betrieb]
Guten Tag, mittlerweile läuft SAE sehr schön, und auch das Schalten des shelly 2 pro PM klappt.
Aber ich hänge an der Routine, den Verbauch zu ermitteln und im sunnyportal anzeigen zu lassen.
Die Konfiguration nach Auslesen aus der log-Datei und nutzen des JSON path finder):
![Bildschirmfoto 2023-02-26 um 16 03 54](https://user-images.githubusercontent.com/126255744/221418882-81ea55f1-3e16-4f7c-a045-ad1f8cc430c5.png)

**Comment by msvintaudio** [comment]
....und da ich nicht weiter schreiben konnte hier die direkte http Ausgabe:
![Bildschirmfoto 2023-02-26 um 16 09 23](https://user-images.githubusercontent.com/126255744/221419140-ecc36a5d-77e0-475c-b4b9-a9b730dd7b15.png)
In der log Datei wird folgendes übermittelt:
![Bildschirmfoto 2023-02-26 um 16 12 24](https://user-images.githubusercontent.com/126255744/221419219-42f1e97a-2782-41d3-bbcf-ba821673541e.png)
Vielen Dank!!

**Comment by camueller** [comment]
Dein Shelly liefert -14 unter `apower`. Scheint kaputt zu sein ...

**Comment by msvintaudio** [comment]
Hatte es mir fast gedacht, da mir das negative Vorzeichen nicht plausibel erschien.
Habe den zweiten Kanal mal getestet, und dort wird auch in der shelly app keine wert angezeigt. Also kaputt!
Das ist für den Start mit der ganzen Geschichte nicht erfreulich.
Farge an der Stelle: was wäre eine Alternative?

**Comment by camueller** [comment]
Bisher habe ich über die Shelly nichts Schlechtes gehört. Ich selbst setzte nur Smart-Plugs mit Tasmota-Fiirmware ein, wie z.B.

https://www.amazon.de/vorgeflasht-stromverbrauch-stromz%C3%A4hler-zeitschaltuhr-NOUS/dp/B0054PSIDW/ref=asc_df_B0054PSIDW/?tag=googshopde-21&linkCode=df0&hvadid=546520626712&hvpos=&hvnetw=g&hvrand=11947343211564493940&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9044435&hvtargid=pla-1604264872253&psc=1&th=1&psc=1

**Reply by sgrafcs** [reply]
Die Dinger werde aber bei etwas höheren Ströme schnell sehr warm, eine Waschmaschine oder Geschirrspüler mit erfassen ist nichts für schwache Nerven.

Meine Hoffnung ist, dass der Shelly Pro da etwas besser ist, er ist ja immerhin für 3.600 Watt zugelassen. Bei mir muss er im Hochsommer nur ca. 1.500 Watt erfassen mal sehen wie warm er da wird.

**Reply by msvintaudio** [reply]
....das habe ich nun anders gelöst, da in der Tat die Ströme am Limit des Shelly gelegen hätten. Ein "normales" Shelly 2 Pro schaltet pro Kanal einen Schütz, der dann den Stromkreis freigibt.
![IMG_1485](https://user-images.githubusercontent.com/126255744/221832633-fea9f5a1-9d6f-422e-a63f-f29f46ffe957.JPG)
Und das dann wirklich komplexere Thema wird es sein, die Umschaltung intelligent zu automatisieren.
Aber da stehe ich völlig am Anfang....

**Comment by msvintaudio** [comment]
Vielen Dank!! Vielleicht war es Zufall, aber das ist bei Erstnutzung nicht so vertrauenerweckend.....

**Comment by sgrafcs** [comment]
Die Shelly Pro kann auch negative Werte :-), ist ganz nette um bei WRs Erzeugung und StandBy-Verbrauch zu ermitteln. 
Aber drauf achten, der Pro vermisst sich dabei etwas, da kommen schon mal 200 mA bei raus, die gar nicht anliegen. Warum habe ich noch nicht rausgefunden.

Das Ganze hängt davon ab wie die Last eingebunden wird. Einfach die beiden Kabel oben an In und Out tauschen.

**Reply by msvintaudio** [reply]
......die Stromumkehr hatte ich auch im Hinterkopf - hatte leider nicht funktioniert.... Die Vermutung, dass das shelly defekt ist, war korrekt, weil nach versuchsweise umstecken auf den 2. Schalter gar nichts angezeigt wurde. Für den Anfang etwas frustrierend, aber gleich etwas gelernt....

**Comment by sgrafcs** [comment]
Mit .apower anstelle vom .power sollt es dann gehen.

![image](https://user-images.githubusercontent.com/32098328/221821893-9f66ab50-1fc6-473d-9eb9-8fa19388521c.png)

**Reply by msvintaudio** [reply]
...und genau das klappte leider nicht.... Also auch mit .apower "kam nichts"....

**Reply by sgrafcs** [reply]
Das passiert bei neu eingerichtete Verbrauchen öfters. Am ersten Tag zeigen die 0 Watt und ab dem zweiten Tag klappt es dann, warum auch immer.

**Comment by msvintaudio** [comment]
.....mittlerweile sind die Tasmota Steckdose, die Alex empfohlen hatte, eingetroffen und installiert. Die erste der 3 installierten Steckdosen lieferte sofort den Verbrauch und lässt sich schön im sunny portal darstellen.
Die zweite Steckdose ist nur noch in der Tagesübersicht - aber nicht in der aktuellen Darstellung.
Die dritte Steckdose ist angelegt - aber keine Verbrauchsdaten....
Dafür sind in der Tasmota App die dargestellten Voltzahlen "Hausnummern".
Mal schauen, wie es sich weiter entwickelt.

**Reply by camueller** [reply]
Tasmota-Adaper sollte man kalibrieren.

**Reply by msvintaudio** [reply]
..ups, das wusste ich nicht.  Inzwischen sehe ich dafür aber in der Tagesansicht ALLE Verbräuche ;-)

**Reply by camueller** [reply]
Siehe https://tasmota.github.io/docs/Power-Monitoring-Calibration/

**Comment by msvintaudio** [comment]
Erledigt - und es passt!! Danke...
Nun geht es an das Thema Steuerung über aktuelle Leistung und Batterieladung. Mal schauen, aber das ist gewiss etwas komplexer.

**Comment by msvintaudio** [comment]
Tja und nun erschliesst sich allmählich wie mächtig das gesamte Werkzeug ist. Mein Respekt Axel, was Du da auf die Beine gestellt hast!!
Mittlerweile läuft der broker und auch rednode, und die Kommunikation zwischen Sunny home manger und SAE klappt.
Der Stellenwert des Zeit/Schaltplans ist mir jetzt auch klar und dank der doc's kaue ich mich langsam durch.
Heute scheint kräftig die Sonne und ich bin gespannt, was passiert, wenn Batterieladung okay ist und die SHM Parameter übertragen werden......
Kann das übrigens auch alles in Ruhe machen, da seid gestern die Rente gezahlt wird ;-)

**Comment by msvintaudio** [comment]
....und nun die nächsten Fragen??
Wenn ich die semp Info des SAE auslese, scheinen die Device Info's nicht komplett zu sein. Ich sehe keine Zeitfenster.....
![Bildschirmfoto 2023-03-02 um 13 14 33](https://user-images.githubusercontent.com/126255744/222426397-2f2fe314-c7c2-4b89-a724-9421055b13e4.png)
Das ist der erste Ausschnitt, wohl die allgemeinen Info's
Dann der zweiter Abschnitt mit der "power" Info
![Bildschirmfoto 2023-03-02 um 13 15 06](https://user-images.githubusercontent.com/126255744/222426569-d6325e36-f2d7-4563-97f3-6be787b13ce3.png)
Aber werden nicht über diese semp Info auch die Schaltzeiten übermittelt??
Oder stehe ich gerade auf dem Schlauch?
Hintergrund: ich bekomme für die 3 Geräte keine Bedarfsanforderung, obwohl Batterie voll und 3 kW Überschuss....

**Comment by msvintaudio** [comment]
...und hier die Schaltplan Konfiguration
![Bildschirmfoto 2023-03-02 um 13 20 02](https://user-images.githubusercontent.com/126255744/222427255-1e7bafc8-1122-4c29-a2ad-acf5e4a8c40d.png)

**Comment by camueller** [comment]
Du hast `externe Aktivierung` angeklickt: Das bedeutet, dass Du Du dem SAE erst eine MQTT-Nachricht schicken musst, damit er die Anforderung (timeframe)  an den SHM übermittelt. Willst Du das wirklich?

Willst Du wirklich, dass er in der ganzen Woche nur 3 Stunden läuft?? Falls Du eigentlich willst, dass er täglich 3 Stunden läuft, musst Du den Tagesplan verwenden.

Im `/semp` findest Du die `timeframe` ganz am Ende!

**Comment by msvintaudio** [comment]
ah, okay!! Das versuche ich gleich mal......

**Comment by msvintaudio** [comment]
...übrigens: Danke für die immer wieder rasche Antwort!!

**Comment by msvintaudio** [comment]
Jtzt klappt es!! Dazu musste ich den Wochenplan löschen, speichern, und dann den korrekten Tagesplan anlegen. Und siehe da: die Ampel steht "automatisch" auf gelb..... Super!!

**Comment by msvintaudio** [comment]
....und nun auf grün, d.h. dass die Anforderung funktioniert. Jetzt muss ich nur noch die Tages-Schalt-Zeiten optimieren, dann macht SAE die Steuerung und die Dame des Hauses braucht nur beladen und das Programm wählen - Fertig!!
Super gut und eine mächtige Erweiterung im Zusammenhang mit off-grid Betrieb, also echten Inselbetrieb für die Hauptlast-Verbraucher.
